presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1125,13 +1125,15 @@ private Regex getConstantPattern(LikePredicate node)
         public Object visitCast(Cast node, Object context)
         {
             Object value = process(node.getExpression(), context);
+            Type type = metadata.getType(parseTypeSignature(node.getType()));
 
-            if (value instanceof Expression) {
-                return new Cast((Expression) value, node.getType(), node.isSafe(), node.isTypeOnly());
+            if (type.equals(type(node.getExpression())) || node.isTypeOnly()) {
```

Comment:
I don't think we can remove type only casts (if the `value` is an `Expression`). I think it's purpose is to preserve type information for the execution and operators.

Also, type only cast might fail (e.g: precision too small).

Reply:
Yeah, for some reason I was thinking they were meaningful only during planning. I reverted that part.